cmake/CompilerOptions.cmake

```diff
@@ -172,7 +172,9 @@ function(set_arch_target optvar arch)
 
         # Use -mcpu for all supported targets _except_ for x86, where it should be -march.
 
-        if(target_model MATCHES "x86|i[3456]86" OR target_model MATCHES "amd64" OR target_model MATCHES "aarch64")
+        if("${target}" MATCHES "aarch64-apple-darwin")
+            set(arch_opt "-mcpu=${arch}")
```

Comment:
Interesting. For `setup.py`, it was setting `arch=armv8-a`, on CMake's level it's `cpu=aarch64`.
If it this produces correct code for you, we will gladly merge it. If you want to double-check that
vectorisation works, I tried this on my machine
```bash
objdump -d lib/libarbor.a | rg 'kamt.*rates.*:' -A 100
```
and looked for AVX instructions. 
There's another diagnostic you can enable, hidden in the `test/unit/CMakeLists.txt`
```cmake
if(${CMAKE_POSITION_INDEPENDENT_CODE})
  make_catalogue(
    NAME dummy
    SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/dummy"
    OUTPUT "CAT_DUMMY_SOURCES"
    MECHS dummy
    PREFIX "${PROJECT_SOURCE_DIR}/mechanisms"
    STANDALONE ON
    VERBOSE ON)  # Set this to ON
 ```
 Your variant will be slightly different, but only the `VERBOSE` setting is relevant.
 Running `make tests` produced this for me
 ```bash
 [0/1] Re-running CMake...
-- PYTHON_EXECUTABLE: /usr/local/Frameworks/Python.framework/Versions/3.9/bin/python3.9
Catalogue name:       dummy
Catalogue mechanisms: dummy
Catalogue sources:    /Users/hater/src/arbor/test/unit/dummy
Catalogue output:     /Users/hater/src/arbor/build/test/unit/generated/dummy
Build as standalone:  ON
Arbor arch:           -march=native
Arbor cxx compiler:
Script prefix:        /Users/hater/src/arbor/mechanisms
Current cxx compiler: /usr/local/opt/llvm/bin/clang++
```
where you can check the `arch` setting.

Reply:
Hi, 

Yes, this PR will install locally without issues as `pip install -v .` on my machine (miniforge conda environment), without specifying `arch` (didn't check builds w. MPI yet). I can run some examples from `python/example` with the build without issues.

As for vectorization,
```
objdump -d lib/libarbor.a | grep 'kamt.*rates.*:' -A 100
```
produces a bunch of output like:
```
00000000000001d0 __ZN3arb17default_catalogue25kernel_mechanism_cpu_kamt6tratesEPNS0_22mechanism_cpu_kamt_pp_Eidd:
     1d0: ed 33 bb 6d                  	stp	d13, d12, [sp, #-80]!
     1d4: eb 2b 01 6d                  	stp	d11, d10, [sp, #16]
     1d8: e9 23 02 6d                  	stp	d9, d8, [sp, #32]
     1dc: f4 4f 03 a9                  	stp	x20, x19, [sp, #48]
     1e0: fd 7b 04 a9                  	stp	x29, x30, [sp, #64]
     1e4: fd 03 01 91                  	add	x29, sp, #64
     1e8: 08 1c a0 4e                  	mov.16b	v8, v0
     1ec: f3 03 00 aa                  	mov	x19, x0
     1f0: 08 b4 40 f9                  	ldr	x8, [x0, #360]
     1f4: 34 7c 7d 93                  	sbfiz	x20, x1, #3, #32
     1f8: 00 69 74 fc                  	ldr	d0, [x8, x20]
     1fc: 02 10 77 1e                  	fmov	d2, #-24.00000000
     200: 21 28 62 1e                  	fadd	d1, d1, d2
     204: 08 00 00 90                  	adrp	x8, #0
     208: 02 01 40 fd                  	ldr	d2, [x8]
     20c: 21 08 62 1e                  	fmul	d1, d1, d2
     210: 00 00 00 94                  	bl	#0 <__ZN3arb17default_catalogue25kernel_mechanism_cpu_kamt6tratesEPNS0_22mechanism_cpu_kamt_pp_Eidd+0x40>
     214: 09 1c a0 4e                  	mov.16b	v9, v0
     218: 60 7e 40 fd                  	ldr	d0, [x19, #248
...
```
Don't know what to look for to be frank. I see no pattern matching "AVX". 

This is what I see during build with the VERBOSE parameter ON
```
...
-- PYTHON_EXECUTABLE: /Users/espenhagen/miniforge3/envs/arbor/bin/python3.9
....
Catalogue name:       dummy
  Catalogue mechanisms: dummy
  Catalogue sources:    /private/var/folders/3r/gg20mbx93rq7b7g2q_4wgl6c0000gn/T/pip-req-build-pqzib_ei/test/unit/dummy
  Catalogue output:     /private/var/folders/3r/gg20mbx93rq7b7g2q_4wgl6c0000gn/T/pip-req-build-pqzib_ei/build/temp.macosx-11.0-arm64-3.9/test/unit/generated/dummy
  Arbor source tree:    /private/var/folders/3r/gg20mbx93rq7b7g2q_4wgl6c0000gn/T/pip-req-build-pqzib_ei
  Build as standalone:  ON
  Arbor arch:           -mcpu=native
  Arbor cxx compiler:
  Current cxx compiler: /Users/espenhagen/miniforge3/envs/arbor/bin/arm64-apple-darwin20.0.0-clang++
...
```
which looks correct w. `Arbor arch:           -mcpu=native`